python/cudf/dataframe/index.py

```diff
@@ -537,6 +561,26 @@ def find_label_range(self, first, last):
             end += 1
         return begin, end
 
+    @property
+    def is_unique(self):
+        return len(set(self._values)) == len(self._values)
+
+    @property
+    def is_monotonic_increasing(self):
```

Comment:
For `DatetimeIndex` specifically (a subclass of `GenericIndex`), can this not be done on the device?